src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/LinkFieldIndexProvider.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using YesSql.Indexes;
+
+namespace OrchardCore.ContentFields.Indexing
+{
+    public class LinkFieldIndex : ContentFieldIndex
+    {
+        public string Url { get; set; }
+        public string Text { get; set; }
+    }
+
+    public class LinkFieldIndexProvider : ContentFieldIndexProvider
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly HashSet<string> _ignoredTypes = new HashSet<string>();
+        private IContentDefinitionManager _contentDefinitionManager;
+
+        public LinkFieldIndexProvider(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public override void Describe(DescribeContext<ContentItem> context)
+        {
+            context.For<LinkFieldIndex>()
+                .Map(contentItem =>
+                {
+                    // Can we safely ignore this content item?
+                    if (_ignoredTypes.Contains(contentItem.ContentType))
+                    {
+                        return null;
+                    }
+
+                    if (!contentItem.Latest && !contentItem.Published)
+                    {
+                        return null;
+                    }
+
+                    // Lazy initialization because of ISession cyclic dependency
+                    _contentDefinitionManager = _contentDefinitionManager ?? _serviceProvider.GetRequiredService<IContentDefinitionManager>();
+
+                    // Search for Text fields
+                    var fieldDefinitions = _contentDefinitionManager
+                        .GetTypeDefinition(contentItem.ContentType)
+                        .Parts.SelectMany(x => x.PartDefinition.Fields.Where(f => f.FieldDefinition.Name == nameof(LinkField)))
+                        .ToArray();
+
+                    var results = new List<LinkFieldIndex>();
+
+                    foreach (var fieldDefinition in fieldDefinitions)
+                    {
+                        var jPart = (JObject)contentItem.Content[fieldDefinition.PartDefinition.Name];
+
+                        if (jPart == null)
+                        {
+                            continue;
+                        }
+
+                        var jField = (JObject)jPart[fieldDefinition.Name];
+
+                        if (jField == null)
+                        {
+                            continue;
+                        }
+
+                        var field = jField.ToObject<LinkField>();
+
+                        if (!String.IsNullOrEmpty(field.Text))
+                        {
+                            results.Add(new LinkFieldIndex
```

Comment:
Looks like you could create an abstract base class as they are all the same.
Also aren't you missing the test that `!Published && !Latest` should return null?

Reply:
Ok working on it but having issues making the base.Describe work.

Reply:
The idea is that in a Describe we do context.For<BooleanFieldIndex>() for example.
But if I want to pass a dynamic XFieldIndex type for the base.Describe(context) then I'm stuck there 😄

Reply:
And if I do context.For<ContentFieldIndex>() then it would require that I have an other db table of that name and that's not the goal.